### DISCUSSION TITLE
Como vocês estão testando suas interfaces?

### DISCUSSION THREAD

**Discussion by aragostinho** [frontendbr/forum#29] [Discussão]
Oi pessoal,
Sou back-end e acredito que na nossa área de desenvolvimento metodologias, técnicas e ferramentas para testes estão bem resolvidas desde a validação de requisitos à testes comportamentais e de integração. Hoje na minha opinião só não pratica quem não quer ou por ainda existir uma grande resistência ou falta de conhecimento.

Para a área de front-end, como vocês estão testando e validando suas interfaces? Principalmente o que diz respeito à cross browser e cross device. Até onde venho estudando em IHC, a validação começa logo nos primeiros passos da engenharia centrada no usuário, seja ela validando necessidades, identificando perfis e documentando  os diversos contextos de uso. Digamos que vocês já tenham feito toda essa a lição de casa e as interfaces criadas agora precisam ser validadas na prática.

Como vocês estão fazendo isso?

**Comment by felipefialho** [comment]
@eduardojmatos  :p

**Comment by eduardojmatos** [comment]
@LFeh opa! hehehe

@aragostinho fiz uma palestra sobre testes de CSS, que no caso, eram também relacionados à interface.

https://speakerdeck.com/eduardojmatos/testes-de-css

Resumindo algumas coisas, a técnica principal usada de teste de interface é usando image diff, que você compara prints de telas em diferentes viewports. 

Cria-se um "mapa de calor" mostrando as diferenças entre cada alteração feita. Tem várias ferramentas que fazem isso. Você pode configurar até o nível de threshold.

**Comment by afonsopacifer** [comment]
@eduardojmatos, vc simplesmente mitou nessa palestra!

....
só isso mesmo, não tenho nada a dizer sobre o assunto proposto rsrsrrsr.

**Comment by mksanti** [comment]
@eduardojmatos Parabens mano. Pelos sliders da pra ver que a palestra foi top.

**Comment by willianjusten** [comment]
Quando alguém conseguir construir um teste de interface mais simple, automático e robusto no nível de outros testes, juro que eu pago um engradado de cerveja para essa pessoa.

**Comment by eduardojmatos** [comment]
@afonsopacifer @mikejavier vlw! São testes simples, mas que precisam de um dev pra acontecer.

@willianjusten falamos disso no FrontUX, se liga: http://frontux.com/testando-interafaces/

**Comment by ghost** [comment]
@eduardojmatos obrigado por compartilhar essas informações, sinceramente eu não me preocupava com todos os testes, até comecei a rever meus projetos :) Você tem alguma palestra gravada pra gente? xD

**Comment by willianjusten** [comment]
Opa @eduardojmatos , eu conheço boa parte dessas ferramentas, inclusive já utilizei. Só que poxa, ainda estão longe de serem tão automáticas como um rspec da vida do Ruby ou unittest do Python...
Mas confio que vai melhorar =)

**Comment by eduardojmatos** [comment]
@meche cara, sobre pré-processador até tenho uma ("o que não te contaram sobre pré-processadores"), mas foge um pouco desse assunto da thread hehehe https://www.youtube.com/watch?v=NcUPeOqG6nY

**Comment by eduardojmatos** [comment]
@willianjusten sim, ainda não há uma ferramenta definitiva pra testes. Até brincaram no FrontUX de planejar e começar a fazer uma hehehe

**Comment by willianjusten** [comment]
@eduardojmatos , se precisar de ajuda, estamos aí =)
